src/os/interaction/modifyinteraction.js

```diff
@@ -409,6 +410,21 @@ class Modify extends OLModify {
     if (this.clone_) {
       const originalGeom = this.clone_.getGeometry();
       if (originalGeom) {
+        // While handling events, OpenLayers may add a new Z (altitude) coordinate to a set of XY
+        // coordinates. Since it is not intended for XY coordinates to have a Z coordinate, these
+        // new values can potentially be undefined and cause problems with Cesium. To prevent this,
+        // set all undefined values to zero.
+        if (originalGeom instanceof Polygon) {
+          var flatCoordinates = originalGeom.getFlatCoordinates();
+          const isUndefined = (item) => item === undefined;
+          if (flatCoordinates && flatCoordinates.some(isUndefined)) {
+            while (flatCoordinates.some(isUndefined)) {
+              flatCoordinates[flatCoordinates.findIndex(isUndefined)] = 0;
+            }
```

Comment:
This loop used to be simple:
`while (flatCoordinates.includes(undefined)) {
  flatCoordinates[flatCoordinates.indexOf(undefined)] = 0;
}`

However, this led to some issues during compile:
`ERROR in ./src/os/interaction/modifyinteraction.js:420 from closure-compiler: actual parameter 1 of Array.prototype.includes does not match formal parameter
found   : undefined
required: number
            while (flatCoordinates.includes(undefined)) {
                                                                 ^^^^^^^^^

ERROR in ./src/os/interaction/modifyinteraction.js:421 from closure-compiler: actual parameter 1 of Array.prototype.indexOf does not match formal parameter
found   : undefined
required: number
              flatCoordinates[flatCoordinates.indexOf(undefined)] = 0;
                                                                                  ^^^^^^^^^`

Since I was trying to find values that were `undefined`, I resorted to these less straight forward functions. Let me know if there is a more elegant way to replace `undefined` values.

Reply:
This loop could be much more efficient. Both `some` and `findIndex` will iterate over the array until the condition is true, which means you're searching for undefined values from the start of the array for each undefined value. Given we have a flat array of coordinates, we at most should need to traverse the array once to replace any undefined values we encounter. We also _only_ want to replace undefined altitude values. We in theory shouldn't have undefined lat/lon, but we also don't want to replace those with 0's because then we're defining a coordinate that's probably not correct.

The fastest solution here would be to:
- Verify the geometry stride is >= 3 so we know it has altitude. The layout could be XYZ or XYZM (has time), but we only need to check that it's at least 3 to have the Z component.
- If stride is >= 3, iterate over the flat coords with a for loop that increments the iterator with `i += stride`, and sets `arr[i + 2] = 0` if the value is not already defined.

This will run in `O(n / stride)` time.